components/profile.js

```diff
@@ -103,15 +102,273 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
+									for (var n = 0; n < 4; n++) {
+										values["profile_showcase[13][" + n + "][publishedfileid]"] = settings[i][type]["values"][n] || "";
+									}
+								}
+								break;
+
+							case 'trade':
+								values["profile_showcase[4]"] = 4;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+
+									if (settings[i][type]["values"].hasOwnProperty("notes")) {
+										values["rgShowcaseConfig[4][6][notes]"] = settings[i][type]["values"]["notes"];
+									}
+
+									if (settings[i][type]["values"].hasOwnProperty("items")) {
+										showcaseconfig.supplied = true;
+										showcaseconfig.numberofrequests = 6;
+										showcaseconfig.showcasetype = 4;
+										showcaseconfig.itemarray = settings[i][type]["values"]["items"];
+									}
+								}
+								break;
+
+							case 'items':
+								values["profile_showcase[3]"] = 3;
+
+								if (settings[i][type].hasOwnProperty("values") && settings[i][type]["values"].hasOwnProperty("items")) {
+									showcaseconfig.supplied = true;
+									showcaseconfig.numberofrequests = 10;
+									showcaseconfig.showcasetype = 3;
+									showcaseconfig.itemarray = settings[i][type]["values"]["items"];
+								}
+								break;
+
+							case 'game':
+								values["profile_showcase[6]"] = 6;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+									values["rgShowcaseConfig[6][0][appid]"] = settings[i][type]["values"];
+								}
+								break;
+
+							case 'badge':
+								values["profile_showcase[5]"] = 5;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+
+									if (settings[i][type]["values"].hasOwnProperty("style")) {
+										var styles = ["rare", "selected", null, "recent", "random"];
+										values["profile_showcase_style_5"] = styles.indexOf(settings[i][type]["values"]["style"]);
+									}
+
+									if (settings[i][type]["values"].hasOwnProperty("badges")) {
+										var types = ["badgeid", "appid", "border_color"];
+										for (var n = 0; n < 6; n++) {
+											for (var m = 0; m < 3; m++){
+												if (settings[i][type]["values"]["badges"][n] != undefined) {
+													if (settings[i][type]["values"]["badges"][n].hasOwnProperty(types[m])){
+														values["rgShowcaseConfig[5][" + n + "][" + types[m] + "]"] = settings[i][type]["values"]["badges"][n][types[m]] || values["rgShowcaseConfig[5][" + n + "][" + types[m] + "]"] || "";
+													}
+												}
+											}
+										}
+									}
+								}
+
+								break;
+
+							case 'rareachievements':
+								values["profile_showcase[1]"] = 1;
+								break;
+
+							case 'screenshot':
+								values["profile_showcase[7]"] = 7;
+
+								if (settings[i][type].hasOwnProperty("values")) {
+									for (var n = 0; n < 4; n++) {
+										if (settings[i][type]["values"][n] != undefined) {
+											values["rgShowcaseConfig[7][" + n + "][publishedfileid]"] = settings[i][type]["values"][n];
```

Comment:
What do you mean? n<4 instead of f.e. `for(number in settings[i][type]["values"])` - I did this because there are 4 items which you can change. So the user can't supply too many now, and if he defines f.e. only 2, the other 2 are left unchanged.

Reply:
I'd still do:

Reply:
So what if the user supplies 5 values. Nothing might happen, something might happen, who knows. Also all the code I saw was without using template strings, let, const, arrow functions, ... so I refrained from using these.

Reply:
I'd say template strings are fair game, because this library only officially supports Node.js 4+ (https://github.com/DoctorMcKay/node-steamcommunity/blob/b7a3e51bfcadd98e342e190f13085268de9801cd/package.json#L34), which also seems a little generous and might be able to be bumped, considering 4.0 is well past its LTS.

@DoctorMcKay, you added your stats reporting library to at least `node-steam-tradeoffer-manager` – what are the percentages of people using various different versions of Node, assuming you record those numbers?

Reply:
@NetroScript The very first word of `profile.js` is `const`. McKay have updated his libs to use templates and arrow functions ([ref](https://github.com/DoctorMcKay/node-steam-tradeoffer-manager/commit/578f3c4aad1a65e9b1e3a7c210f797399425b092)), seems like `steamcommunity` didn't get enough care though, that said, the example was about using `Array.forEach` instead of random loop.

> So what if the user supplies 5 values. Nothing might happen, something might happen, who knows.

Well, the code is meant to be deterministic by nature, when user supplies 5 values, it will send one parameter more - what happens on Steam side is unknown indeed (it'll likely get discarded), if we know there's a restriction of 4 entries - we should somehow prevent the user from supplying more (eg. by throwing an error), instead of coercing the input.

You probably only want to use non descriptive loops when taking some sort of *action* multiple times, eg. sending `n` HTTP requests a few lines below, otherwise it leads to confusion - "Why `4`? What's `settings[i][type]["values"][n]`?"